### PR TITLE
Implemented distance function: distance edit in the UI and distance v…

### PR DESCRIPTION
**Pull request by DominikWas**
## Features
- Added UI for distance under settings
- Allow settings model to accept distance parameter 
- Added validation for settings ( Max 5, Min 1 )

## Bug fixing
None

## Bugs found
None

Tasks:
#31 
#19 
#23 

Note: Task 23 could not replicate

### CODE REVIEW

File: src/app/settings/settings.page.html [hunk 1]
```diff
@@ -35,4 +35,23 @@
         <ion-text *ngIf="settingsForm.get('locInterval').invalid" color="danger"
                   fxFlex="90">{{getFormError('locInterval')}}</ion-text>
     </form>
+    <form [formGroup]="settingsForm" fxLayout="row wrap" fxLayoutAlign="start center">
```
Comment (HarryPi) [inline src/app/settings/settings.page.html]: It appears there are 2 formgroup/form elements here.
- Doesn't this break the submit?
- Has this been tested?
- Why not all under one?
@DominikWas

Comment (HarryPi) [reply src/app/settings/settings.page.html]: Similarly with FxLayout unnecessary if duplication is removed

File: src/app/settings/settings.page.ts [hunk 2]
```diff
@@ -23,10 +23,12 @@ export class SettingsPage implements OnInit {
         console.log(this.userSettings.settings);
         this.settings = this.userSettings.settings;
         this.settingsForm = this._fb.group({
-            locInterval: this._fb.control(this.settings.locInterval, [Validators.required, Validators.minLength(1)])
+            locInterval: this._fb.control(this.settings.locInterval, [Validators.required, Validators.minLength(1)]),
+            distance: this._fb.control(this.settings.distance, [Validators.required, Validators.minLength(1), Validators.maxLength(5)])
```
Comment (HarryPi) [inline src/app/settings/settings.page.ts]: Why is this length validatior? does this work for value? Wouldn't this only give an error if for example the value is 111111 six numbers in length?

File: src/app/settings/settings.page.html (hunk 1 above)
Comment (DominikWas) [reply src/app/settings/settings.page.html]: 2nd form has been removed and the content transferred to the first form

File: src/app/settings/settings.page.ts (hunk 2 above)
Comment (DominikWas) [reply src/app/settings/settings.page.ts]: Seems like the correct validators didn't come up in the latest commit. Some validators like minLength are not necessary, removed and edited in latest commit

File: src/app/settings/settings.page.ts [hunk 3]
```diff
@@ -23,10 +23,15 @@ export class SettingsPage implements OnInit {
         console.log(this.userSettings.settings);
         this.settings = this.userSettings.settings;
         this.settingsForm = this._fb.group({
-            locInterval: this._fb.control(this.settings.locInterval, [Validators.required, Validators.minLength(1)])
+            locInterval: this._fb.control(this.settings.locInterval, [Validators.required,
+                                                                      Validators.maxLength(5)]),
+            distance: this._fb.control(this.settings.distance, [Validators.required,
+                                                                Validators.maxLength(3),
```
Comment (HarryPi) [inline src/app/settings/settings.page.ts]: Why is there a Validator for max length?

Comment (DominikWas) [reply src/app/settings/settings.page.ts]: Removed the maxLength validator, redundant. Added min validator for user input

**Pull request merged**